core/consensus/common/src/lib.rs

```diff
@@ -66,9 +65,8 @@ pub trait Environment<B: Block, ConsensusData> {
 	type Error: From<Error>;
 
 	/// Initialize the proposal logic on top of a specific header. Provide
-	/// the authorities at that header, and a local key to sign any additional
-	/// consensus messages with as well.
-	fn init(&self, parent_header: &B::Header, authorities: &[AuthorityId], sign_with: Arc<ed25519::Pair>)
```

Comment:
It doesn't look like we're using the `sign_with` parameter anywhere and it's specific to ed25519, so I just removed it.

Reply:
It has its history in the rhd/bft interface we had before. It is currently not-working and won't be looking the same anyways. We can reintroduce that if we need to (though I doubt this pattern was good in the first place)